### PR TITLE
Refactor, Log messages and bug fix.

### PR DESCRIPTION
**Pull request by oncleben31**
- Refactor to use `current_forecast` in weather sensor.
- Add messages in logs
- Secure now() timestamp computation (possible bug due to timezone)
- new unique_id
- Add todos to be solved before main release.
- convert wind speed in km/h
- better city list in config_flow
- Manage initial CONF_MODE for weather entity as None

### CODE REVIEW

File: homeassistant/components/meteo_france/__init__.py [hunk 1]
```diff
@@ -54,7 +54,12 @@ async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
 
 async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
     """Set up an Meteo-France account from a config entry."""
-    _LOGGER.warning("SETUP")
+    _LOGGER.info(
+        "Start entities setup for %s (lat=%s|lon=%s)",
+        entry.title,
+        entry.data[CONF_LATITUDE],
+        entry.data[CONF_LONGITUDE],
+    )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/__init__.py]: I don't think we need this

File: homeassistant/components/meteo_france/__init__.py [hunk 2]
```diff
@@ -104,13 +109,27 @@ async def _async_update_data_alert():
 
         if not coordinator_rain.last_update_success:
             raise ConfigEntryNotReady
+    else:
+        _LOGGER.info(
+            "1 hour rain forecast not available. %s (lat=%s|lon=%s) is not in covered zone",
+            entry.title,
+            entry.data[CONF_LATITUDE],
+            entry.data[CONF_LONGITUDE],
+        )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/__init__.py]: We have the city name in the title (I think we should use str repr of Place instead), so let and long are not needed

File: homeassistant/components/meteo_france/__init__.py [hunk 3]
```diff
@@ -129,6 +148,13 @@ async def _async_update_data_alert():
             COORDINATOR_ALERT_ADDED: False,
             COORDINATOR_ALERT: coordinator_alert,
         }
+    else:
+        _LOGGER.info(
+            "Weather alert not available. %s (lat=%s|lon=%s) is not in a France department or Andorre.",
+            entry.title,
+            entry.data[CONF_LATITUDE],
+            entry.data[CONF_LONGITUDE],
+        )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/__init__.py]:
```suggestion
        _LOGGER.info(
            "Weather alert not available: The city %s is not in France or Andorre.",
            entry.title,
        )
```

File: homeassistant/components/meteo_france/__init__.py [hunk 4]
```diff
@@ -104,13 +109,27 @@ async def _async_update_data_alert():
...
     }
 
     department = coordinator_forecast.data.position.get("dept")
+    _LOGGER.debug(
+        "Department correspondig to %s (lat=%s|lon=%s) is %s",
+        entry.title,
+        entry.data[CONF_LATITUDE],
+        entry.data[CONF_LONGITUDE],
+        department,
+    )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/__init__.py]: I don't think we need this

File: homeassistant/components/meteo_france/__init__.py [hunk 5]
```diff
@@ -140,7 +166,12 @@ async def _async_update_data_alert():
 
 async def async_unload_entry(hass: HomeAssistantType, entry: ConfigEntry):
     """Unload a config entry."""
-    _LOGGER.warning("UNLOAD")
+    _LOGGER.info(
+        "Start entities unload for %s (lat=%s|lon=%s)",
+        entry.title,
+        entry.data[CONF_LATITUDE],
+        entry.data[CONF_LONGITUDE],
+    )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/__init__.py]: I don't think we need this

File: homeassistant/components/meteo_france/config_flow.py [hunk 6]
```diff
@@ -82,10 +82,8 @@ async def async_step_cities(self, user_input=None, places=None):
         if places and len(places) > 1 and self.source != SOURCE_IMPORT:
             places_for_form = {}
             for place in places:
-                places_for_form[
-                    _build_place_key(place)
-                ] = f"{place.name} - {place.admin} - {place.country}"
-            _LOGGER.warning(places_for_form)
+                places_for_form[_build_place_key(place)] = f"{place}"
+            _LOGGER.debug(places_for_form)
```
Comment (Quentame) [inline homeassistant/components/meteo_france/config_flow.py]:
```suggestion
```

File: homeassistant/components/meteo_france/sensor.py [hunk 7]
```diff
@@ -46,22 +46,40 @@ async def async_setup_entry(
         if sensor_type == "next_rain":
             if coordinator_rain:
                 entities.append(MeteoFranceRainSensor(sensor_type, coordinator_rain))
+                _LOGGER.info(
+                    "Next rain sensor added for %s.",
+                    coordinator_forecast.data.position["name"],
+                )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/sensor.py]: I don't think we need this

File: homeassistant/components/meteo_france/sensor.py [hunk 8]
```diff
@@ -46,22 +46,40 @@ async def async_setup_entry(
...
                 )
-            )
 
         else:
             entities.append(MeteoFranceSensor(sensor_type, coordinator_forecast))
+            _LOGGER.info(
+                "Sensor %s added for %s.",
+                sensor_type,
+                coordinator_forecast.data.position["name"],
+            )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/sensor.py]: I don't think we need this

File: homeassistant/components/meteo_france/sensor.py [hunk 9]
```diff
@@ -46,22 +46,40 @@ async def async_setup_entry(
...
+                coordinator_alert_data[COORDINATOR_ALERT_ADDED] = True
+                entities.append(
+                    MeteoFranceAlertSensor(
+                        sensor_type, coordinator_alert_data[COORDINATOR_ALERT]
+                    )
+                )
+                _LOGGER.info(
+                    "Weather alert sensor added for %s.",
+                    coordinator_forecast.data.position["dept"],
                 )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/sensor.py]: I don't think we need this

File: homeassistant/components/meteo_france/sensor.py [hunk 10]
```diff
@@ -46,22 +46,40 @@ async def async_setup_entry(
...
-            if coordinator_alert_data[COORDINATOR_ALERT_ADDED]:
-                continue
-            coordinator_alert_data[COORDINATOR_ALERT_ADDED] = True
-            entities.append(
-                MeteoFranceAlertSensor(
-                    sensor_type, coordinator_alert_data[COORDINATOR_ALERT]
+            if coordinator_alert_data:
+                if coordinator_alert_data[COORDINATOR_ALERT_ADDED]:
+                    _LOGGER.info(
+                        "Weather alert sensor skipped for department n°%s. Already added previously.",
```
Comment (Quentame) [inline homeassistant/components/meteo_france/sensor.py]:
```suggestion
                        "Weather alert sensor skipped for department n°%s: already added.",
```

File: homeassistant/components/meteo_france/sensor.py [hunk 11]
```diff
@@ -77,11 +95,12 @@ def __init__(self, sensor_type: str, coordinator: DataUpdateCoordinator):
         self.coordinator = coordinator
         city_name = self.coordinator.data.position["name"]
         self._name = f"{city_name} {SENSOR_TYPES[self._type][ENTITY_NAME]}"
+        self._unique_id = f"{self.coordinator.data.position['lat']}/{self.coordinator.data.position['lon']}-{SENSOR_TYPES[self._type][ENTITY_NAME]}"
```
Comment (Quentame) [inline homeassistant/components/meteo_france/sensor.py]:
```suggestion
        self._unique_id = f"{self.coordinator.data.position['lat']},{self.coordinator.data.position['lon']}_{self._type}"
```

File: homeassistant/components/meteo_france/sensor.py [hunk 12]
```diff
@@ -93,12 +112,21 @@ def state(self):
...
         if len(path) == 3:
-            return data[path[1]][path[2]]
-        return data[path[1]]
+            value = data[path[1]][path[2]]
+        value = data[path[1]]
+
+        # Add unit conversion for wind speed. API is in m/s
+        if self._type == "wind_speed":
+            # convesion in km/h
+            value = round(value * 3.6)
```
Comment (Quentame) [inline homeassistant/components/meteo_france/sensor.py]:
```suggestion
        if self._type == "wind_speed":
            # convert API wind speed from m/s to km/h
            value = round(value * 3.6)
```

File: homeassistant/components/meteo_france/weather.py [hunk 13]
```diff
@@ -55,11 +66,12 @@ def __init__(self, coordinator: DataUpdateCoordinator, mode: str):
         self.coordinator = coordinator
         self._city_name = self.coordinator.data.position["name"]
         self._mode = mode
+        self._unique_id = f"{self.coordinator.data.position['lat']}/{self.coordinator.data.position['lon']}-weather-{mode}"
```
Comment (Quentame) [inline homeassistant/components/meteo_france/weather.py]: The mode can be changed, so it should not be part of unique ID

File: homeassistant/components/meteo_france/weather.py [hunk 14]
```diff
@@ -43,7 +44,17 @@ async def async_setup_entry(
...
+                coordinator, entry.options.get(CONF_MODE, FORECAST_MODE_DAILY),
+            )
+        ],
+        True,
+    )
+    _LOGGER.info(
+        "Weather entity (%s) added for %s.",
+        entry.options.get(CONF_MODE, FORECAST_MODE_DAILY),
+        coordinator.data.position["name"],
     )
```
Comment (Quentame) [inline homeassistant/components/meteo_france/weather.py]: I don't think we need this

File: homeassistant/components/meteo_france/weather.py [hunk 15]
```diff
@@ -84,22 +98,23 @@ def temperature_unit(self):
...
     def humidity(self):
         """Return the humidity."""
-        return self.coordinator.data.forecast[2]["humidity"]
+        return self.coordinator.data.current_forecast["humidity"]
 
     @property
     def wind_speed(self):
         """Return the wind speed."""
-        return self.coordinator.data.forecast[2]["wind"]["speed"]
+        # convert m/s from API in km/h
```
Comment (Quentame) [inline homeassistant/components/meteo_france/weather.py]:
```suggestion
        # convert from API m/s to km/h
```

File: homeassistant/components/meteo_france/weather.py [hunk 16]
```diff
@@ -132,12 +149,12 @@ def forecast(self):
                     }
                 )
         else:
-            today = datetime.utcnow().timestamp()
+            # today = datetime.utcnow().timestamp()
```
Comment (Quentame) [inline homeassistant/components/meteo_france/weather.py]:
```suggestion
```

File: homeassistant/components/meteo_france/weather.py [hunk 17]
```diff
@@ -132,12 +149,12 @@ def forecast(self):
...
             for forecast in self.coordinator.data.daily_forecast:
                 # Can have data of yesterday
-                if forecast["dt"] < today:
-                    _LOGGER.debug("remove_forecast %s %s", self._mode, forecast)
-                    continue
+                # if forecast["dt"] < today:
+                #     _LOGGER.debug("remove_forecast %s %s", self._mode, forecast)
+                #     continue
```
Comment (Quentame) [inline homeassistant/components/meteo_france/weather.py]:
```suggestion
```

**Review by Quentame — CHANGES_REQUESTED** [review]
Nice, but:
- be aware that unique_id cannot use values that can be changed
- log things that the use might ask himself "Why it is happenning like that", if everything is working properly, we don't need to flood the user.

Nice for the `current_forecast` that I forgot.

File: homeassistant/components/meteo_france/__init__.py (hunk 1 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/__init__.py]: OK. I will remove it.

File: homeassistant/components/meteo_france/__init__.py (hunk 2 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/__init__.py]: OK taken into account.

File: homeassistant/components/meteo_france/__init__.py (hunk 4 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/__init__.py]: I prefer to keep it currently. To help test and potential debug with Andorre cities. Could be deleted after the release.

Comment (oncleben31) [reply homeassistant/components/meteo_france/__init__.py]: I've removed the lat/lon

File: homeassistant/components/meteo_france/__init__.py (hunk 3 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/__init__.py]: OK

File: homeassistant/components/meteo_france/__init__.py (hunk 5 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/__init__.py]: OK

File: homeassistant/components/meteo_france/config_flow.py (hunk 6 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/config_flow.py]: OK

File: homeassistant/components/meteo_france/sensor.py (hunk 7 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/sensor.py]: I prefere to keep it to be consistent with the other entites.

File: homeassistant/components/meteo_france/sensor.py (hunk 9 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/sensor.py]: Not sure what part of the block is not needed ? The Log ?

File: homeassistant/components/meteo_france/sensor.py (hunk 11 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/sensor.py]: Yes better

File: homeassistant/components/meteo_france/sensor.py (hunk 12 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/sensor.py]: OK

File: homeassistant/components/meteo_france/weather.py (hunk 13 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/weather.py]: OK I wasn't sure for that.

File: homeassistant/components/meteo_france/weather.py (hunk 15 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/weather.py]: Better

File: homeassistant/components/meteo_france/weather.py (hunk 17 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/weather.py]: OK

File: homeassistant/components/meteo_france/sensor.py (hunk 7 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/sensor.py]: And during test and debug phase could be usefull

File: homeassistant/components/meteo_france/sensor.py (hunk 9 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/sensor.py]: I prefer to keep the log during the fine tuning and test.

File: homeassistant/components/meteo_france/sensor.py (hunk 8 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/sensor.py]: I prefer to keep it during test phase.

File: homeassistant/components/meteo_france/weather.py (hunk 14 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/weather.py]: I prefer to keep it during test phase

**Comment by oncleben31** [comment]
I general I prefer to keep some logs during the fine tuning of the dev.
If you prefer we can put them in debug level.

File: homeassistant/components/meteo_france/const.py [hunk 18]
```diff
@@ -77,7 +77,7 @@
         ENTITY_ICON: "mdi:weather-pouring",
         ENTITY_CLASS: "timestamp",
         ENTITY_ENABLE: True,
-        ENTITY_API_DATA_PATH: "current_forecast:T:value",
+        ENTITY_API_DATA_PATH: "NOT_USED",
```
Comment (Quentame) [inline homeassistant/components/meteo_france/const.py]:
```suggestion
        ENTITY_API_DATA_PATH: None,
```

Forgot this one sorry

File: homeassistant/components/meteo_france/const.py [hunk 19]
```diff
@@ -101,7 +101,7 @@
         ENTITY_ICON: "mdi:weather-cloudy-alert",
         ENTITY_CLASS: None,
         ENTITY_ENABLE: True,
-        ENTITY_API_DATA_PATH: "current_forecast:T:value",
+        ENTITY_API_DATA_PATH: "NOT_USED",
```
Comment (Quentame) [inline homeassistant/components/meteo_france/const.py]:
```suggestion
        ENTITY_API_DATA_PATH: None,
```

File: homeassistant/components/meteo_france/sensor.py (hunk 7 above)
Comment (Quentame) [reply homeassistant/components/meteo_france/sensor.py]: We generally don't log when something is added, but when something is not added.

In the case of add, it's the way we want to use the component.

In the other case, it's kinda "and error case", so we should log why this is happening.

File: homeassistant/components/meteo_france/sensor.py (hunk 9 above)
Comment (Quentame) [reply homeassistant/components/meteo_france/sensor.py]: Yep, the log.

We generally don't log when something is added, but when something is not added, to know the cause why it's not added.

File: homeassistant/components/meteo_france/sensor.py (hunk 8 above)
Comment (Quentame) [reply homeassistant/components/meteo_france/sensor.py]: We generally don't log when something is added, but when something is not added, to know the cause why it's not added.

File: homeassistant/components/meteo_france/weather.py [hunk 20]
```diff
@@ -55,11 +66,12 @@ def __init__(self, coordinator: DataUpdateCoordinator, mode: str):
         self.coordinator = coordinator
         self._city_name = self.coordinator.data.position["name"]
         self._mode = mode
+        self._unique_id = f"{self.coordinator.data.position['lat']}/{self.coordinator.data.position['lon']}-weather"
```
Comment (Quentame) [inline homeassistant/components/meteo_france/weather.py]:
```suggestion
        self._unique_id = f"{self.coordinator.data.position['lat']},{self.coordinator.data.position['lon']}"
```

The unique_id is automatically prefixed by the platform, so we should not add it to unique_id.

Here the unique_id will be `weather.4.58956,73.1856`

**Review by Quentame — CHANGES_REQUESTED** [review]

**Comment by Quentame** [comment]
> If you prefer we can put them in debug level.

OK for the debug, you can resolve concerned comments.
Added details to let you know.

File: homeassistant/components/meteo_france/const.py (hunk 19 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/const.py]: OK

File: homeassistant/components/meteo_france/weather.py (hunk 14 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/weather.py]: So put in debug level

File: homeassistant/components/meteo_france/const.py (hunk 18 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/const.py]: OK

**Comment by oncleben31** [comment]
OK I have put the "addition" logs at debug level.

File: homeassistant/components/meteo_france/weather.py (hunk 20 above)
Comment (oncleben31) [reply homeassistant/components/meteo_france/weather.py]: Are you sure for the `unique_id` ? I have followed the recommandation in the [developer documentation](https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements).

> If a device has a single serial but provides multiple entities, combine the serial with unique identifiers for the entities. For example, if a device measures both temperature and humidity, you can uniquely identify the entities using {serial}-{sensor_type}.

Comment (Quentame) [reply homeassistant/components/meteo_france/weather.py]: Extremly sure, I got a comment on a PR about that.

Comment (Quentame) [reply homeassistant/components/meteo_france/weather.py]: about the doc, it's in the case you have multiple type of entities in a same platform

Comment (oncleben31) [reply homeassistant/components/meteo_france/weather.py]: done

**Review by Quentame — APPROVED** [review]

**Pull request merged**